pkg/cluster/operation/destroy.go

```diff
@@ -183,10 +186,26 @@ func DestroyComponent(getter ExecutorGetter, instances []meta.Instance, timeout
 		// Stop by systemd.
 		delPaths := make([]string, 0)
 		switch name {
-		case meta.ComponentTiKV, meta.ComponentPD, meta.ComponentPump, meta.ComponentDrainer, meta.ComponentPrometheus, meta.ComponentAlertManager, meta.ComponentDMMaster, meta.ComponentDMWorker, meta.ComponentDMPortal:
-			delPaths = append(delPaths, ins.DataDir())
+		case meta.ComponentTiKV,
+			meta.ComponentPD,
+			meta.ComponentPump,
+			meta.ComponentDrainer,
+			meta.ComponentPrometheus,
+			meta.ComponentAlertManager,
+			meta.ComponentDMMaster,
+			meta.ComponentDMWorker,
+			meta.ComponentDMPortal:
+			if cls.CountDir(ins.GetHost(), ins.DataDir()) == 1 {
```

Comment:
```suggestion
			if cls.CountDir(ins.GetHost(), ins.DataDir()) <= 1 {
```

Reply:
No it should not process when count is 0, it doesn't make any sense to return 0 and there must be something wrong in the counting function.